src/System.Data.Common/src/System/Data/Common/AdapterUtil.cs

```diff
@@ -0,0 +1,206 @@
+        {
+            return Argument(Res.GetString(Res.ADP_ConnectionStringSyntax, index));
+        }
+        static internal ArgumentException KeywordNotSupported(string keyword)
+        {
+            return Argument(Res.GetString(Res.ADP_KeywordNotSupported, keyword));
+        }
+        static internal ArgumentException InvalidKeyname(string parameterName)
+        {
+            return Argument(Res.GetString(Res.ADP_InvalidKey), parameterName);
+        }
+        static internal ArgumentException InvalidValue(string parameterName)
+        {
+            return Argument(Res.GetString(Res.ADP_InvalidValue), parameterName);
+        }
+        static internal ArgumentException ConvertFailed(Type fromType, Type toType, Exception innerException)
+        {
+            return ADP.Argument(Res.GetString(Res.SqlConvert_ConvertFailed, fromType.FullName, toType.FullName), innerException);
+        }
+        internal enum InternalErrorCode
+        {
+            UnpooledObjectHasOwner = 0,
+            UnpooledObjectHasWrongOwner = 1,
+            PushingObjectSecondTime = 2,
+            PooledObjectHasOwner = 3,
+            PooledObjectInPoolMoreThanOnce = 4,
+            CreateObjectReturnedNull = 5,
+            NewObjectCannotBePooled = 6,
+            NonPooledObjectUsedMoreThanOnce = 7,
+            AttemptingToPoolOnRestrictedToken = 8,
+            //          ConnectionOptionsInUse                                  =  9,
+            ConvertSidToStringSidWReturnedNull = 10,
+            //          UnexpectedTransactedObject                              = 11,
+            AttemptingToConstructReferenceCollectionOnStaticObject = 12,
+            AttemptingToEnlistTwice = 13,
+            CreateReferenceCollectionReturnedNull = 14,
+            PooledObjectWithoutPool = 15,
+            UnexpectedWaitAnyResult = 16,
+            SynchronousConnectReturnedPending = 17,
+            CompletedConnectReturnedPending = 18,
+
+            NameValuePairNext = 20,
+            InvalidParserState1 = 21,
+            InvalidParserState2 = 22,
+            InvalidParserState3 = 23,
+
+            InvalidBuffer = 30,
+
+            UnimplementedSMIMethod = 40,
+            InvalidSmiCall = 41,
+
+            SqlDependencyObtainProcessDispatcherFailureObjectHandle = 50,
+            SqlDependencyProcessDispatcherFailureCreateInstance = 51,
+            SqlDependencyProcessDispatcherFailureAppDomain = 52,
+            SqlDependencyCommandHashIsNotAssociatedWithNotification = 53,
+
+            UnknownTransactionFailure = 60,
+        }
+        static internal Exception InternalError(InternalErrorCode internalError)
+        {
+            return InvalidOperation(Res.GetString(Res.ADP_InternalProviderError, (int)internalError));
+        }
+        internal const int DefaultConnectionTimeout = DbConnectionStringDefaults.ConnectTimeout;
+
+        static internal bool IsEmpty(string str)
```

Comment:
I changed the implementation to use the string.IsNullOrEmpty. 
The code is quite old and this kind of code has been around. Thanks for pointing this out.